compilers/concrete-compiler/compiler/include/concretelang/Dialect/Concrete/Analysis/MemoryUsage.h

```diff
@@ -0,0 +1,64 @@
+// Part of the Concrete Compiler Project, under the BSD3 License with Zama
+// Exceptions. See
+// https://github.com/zama-ai/concrete-compiler-internal/blob/main/LICENSE.txt
+// for license information.
+
+#ifndef CONCRETELANG_DIALECT_CONCRETE_MEMORY_USAGE_H
+#define CONCRETELANG_DIALECT_CONCRETE_MEMORY_USAGE_H
+
+#include <mlir/IR/BuiltinOps.h>
+#include <mlir/IR/Operation.h>
+#include <mlir/Pass/Pass.h>
+
+#include <concretelang/Support/CompilationFeedback.h>
```

Comment:
So here and same reflexion for statistics, why not follow the common pass pattern?